### PR TITLE
#2213 Fixed `ForeignTojos.contains()` method and added test

### PR DESCRIPTION
**Pull request by maxonfjvipon**
Closes: #2213

---

## PR-Codex overview
This PR focuses on fixing a bug in the `RegisterMojo` class and adding tests for the `ForeignTojos` class. 

### Detailed summary
- In `RegisterMojo`, the condition in the `if` statement is reversed.
- In `ForeignTojos`, the `contains` method is modified to return the opposite result.

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Comment by maxonfjvipon** [comment]
@volodya-lombrozo please review

File: eo-maven-plugin/src/test/java/org/eolang/maven/tojos/ForeignTojosTest.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+final class ForeignTojosTest {
+
+    /**
+     * Testable foreign tojos.
+     */
+    private ForeignTojos tojos;
+
+    @BeforeEach
+    void setUp(@TempDir final Path tmp) {
+        this.tojos = new ForeignTojos(() -> new FakeMaven(tmp).foreign());
```
Comment (volodya-lombrozo) [inline eo-maven-plugin/src/test/java/org/eolang/maven/tojos/ForeignTojosTest.java:51]: @maxonfjvipon I don't like the idea of creating the instance of `ForeignTojos` through the filesystem usage. Can we create a `ForeignTojos` instance somehow in a memory, without creating files on the filesystem? 
If it is impossible, please, add new puzzle for it.

File: eo-maven-plugin/src/test/java/org/eolang/maven/tojos/ForeignTojosTest.java [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    }
+
+    @ParameterizedTest
+    @CsvSource({
+        "abs",
+        "org.eolang.int",
+        "QQ.io.stdout"
+    })
+    void contains(final String name) {
```
Comment (volodya-lombrozo) [inline eo-maven-plugin/src/test/java/org/eolang/maven/tojos/ForeignTojosTest.java:60]: @maxonfjvipon You check only the "successful" path. Could you check the negative path (when we can't find a tojo)?

**Review by volodya-lombrozo — CHANGES_REQUESTED** [review]

File: eo-maven-plugin/src/test/java/org/eolang/maven/tojos/ForeignTojosTest.java [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+        this.tojos = new ForeignTojos(() -> new FakeMaven(tmp).foreign());
+    }
+
+    @ParameterizedTest
+    @CsvSource({
+        "abs, abs, true",
+        "org.eolang.int, org.eolang.float, false",
+        "QQ.io.stdout, QQ.io.stdout, true"
+    })
+    void contains(final String name, final String check, final boolean contains) {
```
Comment (volodya-lombrozo) [inline eo-maven-plugin/src/test/java/org/eolang/maven/tojos/ForeignTojosTest.java:68]: @maxonfjvipon Maybe it's better to create a separate test for the  "not contains" case? 
Actually, what you are doing looks like the anti-pattern "The Free Ride / Piggyback":
https://stackoverflow.com/questions/333682/unit-testing-anti-patterns-catalogue/333883#333883

**Review by volodya-lombrozo — CHANGES_REQUESTED** [review]

File: eo-maven-plugin/src/test/java/org/eolang/maven/tojos/ForeignTojosTest.java (hunk 3 above)
Comment (maxonfjvipon) [reply eo-maven-plugin/src/test/java/org/eolang/maven/tojos/ForeignTojosTest.java:68]: @volodya-lombrozo yeah, thanks, you're right

**Comment by maxonfjvipon** [comment]
@volodya-lombrozo please have a look one more time

File: eo-maven-plugin/src/test/java/org/eolang/maven/tojos/ForeignTojosTest.java [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+    }
+
+    @ParameterizedTest
+    @CsvSource({
+        "abs, sba",
+        "org.eolang.int, org.eolang.float",
+        "QQ.io.stdout, QQ.txt.sprintf"
+    })
+    void doesNotContain(final String name, final String check) {
+        this.tojos.add(name);
```
Comment (volodya-lombrozo) [inline eo-maven-plugin/src/test/java/org/eolang/maven/tojos/ForeignTojosTest.java:83]: @maxonfjvipon Do you need to add elements to `tojos` for that test? If you wan't to emphasise that `tojos` already has elements (tojos is not empty) and you are looking among them then I would recommend to init them inside `setUp` function.

Or if you want to leave the test as is, just give more meaningful names for the `name` and `check` variables (maybe something like `name` -> `existing` and `check` -> `considered`)

**Review by volodya-lombrozo — CHANGES_REQUESTED** [review]
@maxonfjvipon One more small comment

File: eo-maven-plugin/src/test/java/org/eolang/maven/tojos/ForeignTojosTest.java (hunk 4 above)
Comment (maxonfjvipon) [reply eo-maven-plugin/src/test/java/org/eolang/maven/tojos/ForeignTojosTest.java]: @volodya-lombrozo I preferred more meangfull names

**Review by volodya-lombrozo — APPROVED** [review]
@maxonfjvipon Looks good to me. Thanks.
@yegor256 Could you merge that changes, please?

**Comment by maxonfjvipon** [comment]
@yegor256 reminder

**Comment by yegor256** [comment]
@rultor merge
